src/net.cpp

```diff
@@ -113,6 +113,32 @@ void CConnman::AddAddrFetch(const std::string& strDest)
 
 uint16_t GetListenPort()
 {
+    // If -bind= is provided with ":port" part, use that (first one if multiple are provided).
+    for (const std::string& bind_arg : gArgs.GetArgs("-bind")) {
+        constexpr uint16_t dummy_port = 0;
+        constexpr bool allow_lookup = false;
+        CService bind_addr;
+
+        if (Lookup(bind_arg, bind_addr, dummy_port, allow_lookup)) {
+            if (bind_addr.GetPort() != dummy_port) {
+                return bind_addr.GetPort();
+            }
+        }
+    }
+
+    // Otherwise, if -whitebind= without PF_NOBAN is provided, use that
+    // (-whitebind= is required to have ":port").
+    for (const std::string& whitebind_arg : gArgs.GetArgs("-whitebind")) {
+        NetWhitebindPermissions whitebind;
+        bilingual_str error;
+        if (NetWhitebindPermissions::TryParse(whitebind_arg, whitebind, error)) {
+            if (!(whitebind.m_flags & PF_NOBAN)) {
```

Comment:
39b4a0c maybe
```suggestion
            if (!NetPermissions::HasFlag(whitebind.m_flags, NetPermissionFlags::PF_NOBAN)) {
```

Reply:
Created a specific `HasAnyFlag` method for this in #21506 at https://github.com/bitcoin/bitcoin/pull/21506/commits/26ac2efcf0acf9bf635f84a9ed986b3d2ddb6042#diff-7016468d59f51adee65131da31d7bebff9818b4e23bb0fac7ab69431b6be4c81R55